features/admin.workflow-models.v1 /components/create/configuration-details-form.tsx

```diff
@@ -0,0 +1,153 @@
+import { PlusIcon } from "@oxygen-ui/react-icons";
+import React, {
+    ForwardedRef,
+    forwardRef,
+    ForwardRefExoticComponent,
+    ReactElement,
+    RefAttributes,
+    useCallback,
+    useEffect,
+    useImperativeHandle,
+    useRef,
+    useState
+} from "react";
+import { RolesInterface } from "@wso2is/core/models";
+import Button from "@oxygen-ui/react/Button";
+import "./configuration-details-form.scss";
+import Box from "@oxygen-ui/react/Box";
+import { UserBasicInterface } from "@wso2is/admin.core.v1/models/users";
+import ApprovalStep from "./approval-step";
+import { ConfigurationsFormValuesInterface } from "../../models/ui";
+
+export interface ConfigurationsPropsInterface {
+    isReadOnly?: boolean;
+    onSubmit?: (values: ConfigurationsFormValuesInterface) => void;
+    initialValues?: Partial<ConfigurationsFormValuesInterface>;
+}
+
+export interface ConfigurationsFormRef {
+    triggerSubmit: () => void;
+}
+
+export type MultiStepApprovalTemplate = {
+    id: string;
+    stepNumber: number;
+    roles: string[];
+    users: string[];
+};
+
+const ConfigurationsForm: ForwardRefExoticComponent<RefAttributes<ConfigurationsFormRef> &
+    ConfigurationsPropsInterface> = forwardRef(
+    ({ isReadOnly, onSubmit, initialValues }: ConfigurationsPropsInterface, ref: ForwardedRef<ConfigurationsFormRef>): ReactElement => {
+        const triggerFormSubmit = useRef<() => void>();
+
+        useEffect(() => {
+            if (initialValues?.approvalSteps?.length) {
+                const parsedSteps = initialValues.approvalSteps.map((step, index) => ({
+                    id: `step${Date.now()}-${index}`,
+                    stepNumber: index + 1,
+                    roles: step.roles || [],
+                    users: step.users || []
+                }));
+        
+                setSteps(parsedSteps);
+                console.log("Initial Values",initialValues)
+                temporarySteps.current.approvalSteps = initialValues.approvalSteps;
+            }
+        }, [initialValues]);
+        
+
+        // Exposing triggerSubmit method
+        useImperativeHandle(ref, () => ({
+            triggerSubmit: () => {
+                const configurationData: ConfigurationsFormValuesInterface = {
+                    approvalSteps: temporarySteps.current.approvalSteps
+                }
+                onSubmit?.(configurationData);
+                // if (triggerFormSubmit.current) {
+                //     triggerFormSubmit.current();
+                // }
+            }
+        }));
+
+        const [steps, setSteps] = useState<MultiStepApprovalTemplate[]>([]);
+        const temporarySteps = useRef<ConfigurationsFormValuesInterface>({
+            approvalSteps: []
+        });
```

Comment:
Move the `useState` and `useRef` items to the top of the file. The order should be as follows. Check the other places as well.

1. Constants
2. Hooks
3. States (useState)
4. useMemo
5. useEffect
6. TS functions
7. UI component